php/libraries/FilesDownloadHandler.php

```diff
@@ -0,0 +1,114 @@
+<?php declare(strict_types=1);
+
+namespace LORIS;
+
+use \Psr\Http\Message\ResponseInterface;
+use \Psr\Http\Message\ServerRequestInterface;
+use \Psr\Http\Server\RequestHandlerInterface;
+
+/**
+ * This class handles files being downloaded from LORIS. It should
+ * serve as the ONLY way that files are downloaded so that all file downloading
+ * functionality can occur on a well-tested foundation.
+ *
+ * @license http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ */
+class FilesDownloadHandler implements RequestHandlerInterface
+{
+
+    const ERROR_EMPTY_FILENAME = 'Invalid filename: cannot be empty';
+    const ERROR_FILE_NOT_FILE  = 'File requested is not a file';
+    /**
+     * The target download directory.
+     *
+     * @var \SplFileInfo
+     */
+    protected $downloadDirectory;
+
+    /**
+     * Create new instance of a File Downloader.
+     *
+     * @param \SplFileInfo $downloadDirectory The target download directory
+     */
+    public function __construct(\SplFileInfo $downloadDirectory)
+    {
+        if (! $downloadDirectory->isDir()) {
+            throw new \LorisException(
+                sprintf(
+                    "Download directory %s is not a directory",
+                    htmlentities($downloadDirectory->getPathname())
+                )
+            );
+        }
+
+        if (! $downloadDirectory->isReadable()) {
+            throw new \LorisException(
+                sprintf(
+                    "Download directory %s is not readable",
+                    htmlentities($downloadDirectory->getPathname())
+                )
+            );
+        }
+        $this->downloadDirectory = $downloadDirectory;
+    }
+
+    /**
+     * Given an HTTP request, serve the file to the client.
+     *
+     * All files uploaded will get the same permissions.
```

Comment:
Unless I am missing something, I don't think this comment applies to this function. This is a downloader, not an uploader.